packages/driver/cypress/integration/e2e/multi-domain/multi_domain_uncaught_errors_spec.ts

```diff
@@ -0,0 +1,175 @@
+        expect(err.message).to.include('The following error originated from your application code, not from Cypress.')
+        expect(err.message).to.not.include('https://on.cypress.io/uncaught-exception-from-application')
+        // @ts-ignore
+        expect(err.docsUrl).to.deep.eq(['https://on.cypress.io/uncaught-exception-from-application'])
+
+        expect(uncaughtExceptionSpy.notCalled).to.be.true
+        expect(runnable === r).to.be.true
+      })
+
+      cy.switchToDomain('foobar.com', () => {
+        cy.get('.trigger-async-error').click()
+
+        // add the cy.wait here to keep commands streaming in,
+        // forcing the switchToDomain callback window to be open long enough for an async error to occur
+        cy.wait(1000)
+      })
+    })
+
+    it('passes the current test/command if async errors are thrown from the switchToDomain context, but the callback window is now closed', () => {
+      const uncaughtExceptionSpy = cy.spy()
+      const failureSpy = cy.spy()
+
+      cy.on('uncaught:exception', uncaughtExceptionSpy)
+
+      cy.on('fail', failureSpy)
+
+      cy.switchToDomain('foobar.com', () => {
+        // the async error here should be thrown AFTER the current command and test has finished, resulting in a passed test with no fail being triggered in the primary
+        cy.get('.trigger-async-error').click()
+      }).then(() => {
+        expect(uncaughtExceptionSpy.notCalled).to.be.true
+        expect(failureSpy.notCalled).to.be.true
+      })
+    })
+  })
+
+  describe('unhandled rejections', () => {
+    it('unhandled rejection triggers uncaught:exception and has promise as third argument', (done) => {
+      cy.switchToDomain('foobar.com', done, () => {
+        const r = cy.state('runnable')
+
+        cy.once('uncaught:exception', (err, runnable, promise) => {
+          expect(err.stack).to.include('promise rejection')
+          expect(err.stack).to.include('one')
+          expect(err.stack).to.include('two')
+          expect(err.stack).to.include('three')
+          expect(runnable === r).to.be.true
+          expect(promise).to.be.a('promise')
+
+          done()
+
+          return false
+        })
+
+        cy.get('.trigger-unhandled-rejection').click()
+      })
+    })
+
+    it('original error is not mutated for "unhandledrejection"', () => {
+      cy.switchToDomain('foobar.com', () => {
+        cy.once('uncaught:exception', () => false)
+
+        cy.get('.trigger-unhandled-rejection').click()
+        cy.get('.error-one').invoke('text').should('equal', 'promise rejection')
+        cy.get('.error-two').invoke('text').should('equal', 'promise rejection')
+      })
+    })
+  })
+
+  it('does not fail if thrown custom error with readonly name', (done) => {
+    cy.once('fail', (err) => {
+      // TODO: we cannot currently serialize custom user errors back to the primary
+      // ideally, we want the name of this error to be 'CustomError'
+      expect(err.name).to.include('Error')
```

Comment:
Why does the name not end up being `CustomError`? Don't we pull off the name property [when serializing it](https://github.com/cypress-io/cypress/blob/22f4f6d9cf494d485137a2c8b2b1a964e3b9f52c/packages/driver/src/multi-domain/communicator.ts#L21)?

Reply:
We do not. We are only merging properties that aren't defined on the newly created error so we get the correct constructor , `toJSON`, and a few other properties. I am wondering if we are going to have to allow list certain properties ( like `name` and `message` ) coming back from the secondary domain, but I was hoping to come up with something a bit more intuitive.

Reply:
I'm gonna play around with some ideas

Reply:
wound up doing something like `reifiedError.name = serializedError?.name ?? reifiedError.name`. Seems to go nicely in there since we are already replacing the stack 2ccc6be0f3dcd3aa8ef04dd44863386b3e11d8b3.